delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/UploadingFile.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations
+
+import akka.http.scaladsl.model.{BodyPartEntity, ContentType}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.UploadedFileInformation
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.Storage.{DiskStorage, RemoteDiskStorage, S3Storage}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.{AbsolutePath, DigestAlgorithm, Storage}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageFileRejection.SaveFileRejection
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageFileRejection.SaveFileRejection.FileContentLengthIsMissing
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Label, ProjectRef}
+
+/**
+  * Represents a file being uploaded with one implementing by storage type
+  */
+sealed trait UploadingFile extends Product with Serializable {
```

Comment:
This allows not to pass all the storage down when we need only a small part of it